tools/SetupFlow/DevHome.SetupFlow/Models/RepositoryProvider.cs

```diff
@@ -271,91 +277,118 @@ public IEnumerable<IDeveloperId> GetAllLoggedInAccounts()
+                }
             }
         }
         catch (AggregateException aggregateException)
         {
             // Because tasks can be canceled DevHome should emit different logs.
             if (aggregateException.InnerException is OperationCanceledException)
             {
-                GlobalLog.Logger?.ReportInfo($"Get Repos operation was cancalled.");
+                Log.Logger?.ReportInfo(Log.Component.RepoConfig, $"Get Repos operation was cancalled.");
             }
             else
             {
-                GlobalLog.Logger?.ReportError($"{aggregateException}");
+                Log.Logger?.ReportInfo(Log.Component.RepoConfig, aggregateException.ToString());
             }
-
-            _repositories[developerId] = new List<IRepository>();
-            return _repositories[developerId];
         }
+        catch (Exception ex)
+        {
+            Log.Logger?.ReportError(Log.Component.RepoConfig, $"Could not get repositories.  Message: {ex}");
+        }
+
+        _repositories[developerId] = repoSearchInformation.Repositories;
+
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_SearchForRepos_Event", LogLevel.Critical, new GetReposEvent("FoundRepos", _repositoryProvider.DisplayName, developerId));
+        return repoSearchInformation;
+    }
 
-        if (result.Result.Status == ProviderOperationStatus.Success)
+    public RepositorySearchInformation GetAllRepositories(IDeveloperId developerId)
+    {
+        TelemetryFactory.Get<ITelemetry>().Log("RepoTool_GetAllRepos_Event", LogLevel.Critical, new GetReposEvent("CallingExtension", _repositoryProvider.DisplayName, developerId));
+        var repoSearchInformation = new RepositorySearchInformation();
+        try
         {
-            if (_repositories.TryGetValue(developerId, out var _))
+            var result = _repositoryProvider.GetRepositoriesAsync(developerId).AsTask().Result;
+            if (result.Result.Status == ProviderOperationStatus.Success)
             {
-                _repositories[developerId] = result.Repositories;
+                repoSearchInformation.Repositories = result.Repositories;
             }
             else
             {
-                _repositories.Add(developerId, result.Repositories);
+                Log.Logger?.ReportError(Log.Component.RepoConfig, $"Could not get repositories.  Message: {result.Result.DisplayMessage}", result.Result.ExtendedError);
             }
         }
-        else
+        catch (AggregateException aggregateException)
         {
-            if (_repositories.TryGetValue(developerId, out var _))
+            // Because tasks can be canceled DevHome should emit different logs.
+            if (aggregateException.InnerException is OperationCanceledException)
             {
-                _repositories[developerId] = new List<IRepository>();
+                GlobalLog.Logger?.ReportInfo($"Get Repos operation was cancalled.");

```

Comment:
not blocking but we could probably use `Log.Logger?.ReportError` here and below instead of the global logger. Though I guess it doesn't matter since all logging in Dev Home will change soon

Reply:
I'll leave it this way for now.